app/controllers/api/v2/provisioning_templates_controller.rb

```diff
@@ -155,6 +161,16 @@ def action_permission
             super
         end
       end
+
+      def find_global_registration
```

Comment:
I'd say this logic should live as a class method of `ProvisioningTemplate`, returning either the object or nil

the render part then belongs to the global_registration, that must verify the template was found